CHANGELOG.md

```diff
@@ -1,66 +0,0 @@
-
```

Comment:
Committed `CHANGELOG.txt` files were indeed standard in the days of mercurial & subversion, but are becoming less common. A few reasons I can think of:

- adhering to [SSoT](https://en.wikipedia.org/wiki/Single_source_of_truth)/[DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principles, i.e. no duplication across `CHANGELOG.txt` and `github.com/OWNER/REPO/releases`
  + btw they were out of sync in this repo - I had to update some of the release notes & add a few missing releases in https://github.com/paskino/qt-elements/releases
- not bloating repo size & reducing download time
- triggering CI
  + GHA can run workflows [`on.release`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release)
  + it's far less common & more complex/error-prone to use [`on.push.paths`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-only-when-a-push-affects-specific-files)
- webscrapers, automated parsing, webhooks & external tools
  + e.g. zenodo.org picks up changelogs from GH releases' notes rather than changelog files

Meanwhile, for progressively building pre-release notes, you could use one of the following places:

1. pull request descriptions
2. commit messages
3. [draft (unpublished) releases](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository#about-release-management)

Note that (for the lazy) the latter 2 options are combined as per [this](https://github.com/paskino/qt-elements/pull/77#discussion_r1254809649)[^gh-autogen-notes], though personally I find it best to put effort into the first option[^pr-release-notes].

Always happy to discuss alternatives though :)

[^gh-autogen-notes]: which is a lightweight version of [GH's verbose autogenerated notes](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes)
[^pr-release-notes]: e.g. https://github.com/NiftyPET/NIMPA/pull/28#issue-1732872191 -> https://github.com/NiftyPET/NIMPA/releases/tag/v2.6.0